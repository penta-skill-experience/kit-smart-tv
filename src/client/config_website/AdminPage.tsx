import * as React from "react";
import Box from '@mui/material/Box';
import Button from '@mui/material/Button';
import {Grid} from "@mui/material";
import InputLabel from "@mui/material/InputLabel";
import Input from "@mui/material/Input";
import InputAdornment from "@mui/material/InputAdornment";
import IconButton from "@mui/material/IconButton";
import VisibilityOff from "@mui/icons-material/VisibilityOff";
import Visibility from "@mui/icons-material/Visibility";
import FormControl from "@mui/material/FormControl";
import Snackbar from "@mui/material/Snackbar";

export const AdminPage = ({oldPassword, newPassword, handleOldPassword, handleNewPassword, handlePasswordChange, children} ) => {
    const [showOldPassword, setShowOldPassword] = React.useState(false);
    const [showNewPassword, setShowNewPassword] = React.useState(false);

    const handleShowOldPassword  = () => {
        setShowOldPassword(!showOldPassword)
    }

    const handleShowNewPassword  = () => {
        setShowNewPassword(!showNewPassword)
    }

    const [successfulBar, setSuccessfulBar] = React.useState(false);
    const [errorBar, setErrorBar] = React.useState(false);

    const handleClose = (event: React.SyntheticEvent | Event, reason?: string) => {
        if (reason === 'clickaway') {
            return;
        }
        setSuccessfulBar(false);
        setErrorBar(false);
    }

    return (
        <Box
        sx={{
            width: 300,
            height: 300,
            }
        }
        >
            <div>
                <Grid container spacing={2} direction="row" justifyContent="center" alignItems="center">
                    <Grid item xs={12}>
                        <FormControl sx={{ m: 1, width: '25ch' }} variant="standard">
                            <InputLabel>Old Password</InputLabel>
                            <Input
                                type={showOldPassword ? 'text' : 'password'}
                                value={oldPassword}
                                onChange={handleOldPassword}
                                endAdornment={
                                    <InputAdornment position="end">
                                        <IconButton
                                            onClick={handleShowOldPassword}
                                        >
                                            {showOldPassword ? <VisibilityOff /> : <Visibility />}
                                        </IconButton>
                                    </InputAdornment>
                                }
                            />
                        </FormControl>
                    </Grid>
                    <Grid item xs={12}>
                        <FormControl sx={{ m: 1, width: '25ch' }} variant="standard">
                            <InputLabel>New Password</InputLabel>
                                <Input
                                type={showNewPassword ? 'text' : 'password'}
                                value={newPassword}
                                onChange={handleNewPassword}
                                endAdornment={
                                    <InputAdornment position="end">
                                        <IconButton
                                        onClick={handleShowNewPassword}
                                        >
                                        {showNewPassword ? <VisibilityOff /> : <Visibility />}
                                        </IconButton>
                                    </InputAdornment>
                                }
                                />
                        </FormControl>
                    </Grid>
                    <Grid item xs={12}>
                        <Button
                            onClick={() => {
                                if (handlePasswordChange()) {
                                    setSuccessfulBar(true);
                                } else{
                                    setErrorBar(true);
                                }
                            }}
                            variant="outlined"
                        >
                            Save Password
                        </Button>
                        <Snackbar
                            open={successfulBar}
                            autoHideDuration={6000}
                            onClose={handleClose}
                            message={'Password Saved'}
                        />
                        <Snackbar
                            open={errorBar}
                            autoHideDuration={6000}
                            onClose={handleClose}
                            message={'New Password must at least be 1 character long'}
                        />
                    </Grid>
                    <Grid item xs={12}>
                    </Grid>
                </Grid>
            </div>
        </Box>
    );
}