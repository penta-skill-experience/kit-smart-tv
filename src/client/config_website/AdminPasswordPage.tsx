import * as React from "react";
import Box from '@mui/material/Box';
import TextField from '@mui/material/TextField';
import Button from '@mui/material/Button';

export const AdminPasswordPage = ({oldPassword, newPassword, handleOldPassword, handleNewPassword, handlePasswordChange, children} ) => {

    return (
        <Box
        sx={{
            width: 300,
            height: 300,
            }
        }
        >
            <div>
                <TextField
                    value={oldPassword}
                    label="Old Password"
                    onChange={handleOldPassword}
                />
                <TextField
                    value={newPassword}
                    label="New Password"
                    onChange={handleNewPassword}
                />
                <Button
                    onClick={handlePasswordChange}
                    variant="outlined"
                >
                    Save Password</Button>
            </div>
        </Box>
    );

}