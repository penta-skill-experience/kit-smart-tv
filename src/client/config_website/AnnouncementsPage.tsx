import * as React from "react";
import {Grid} from "@mui/material";
import TextField from '@mui/material/TextField';
import Button from "@mui/material/Button";
import DeleteIcon from "@mui/icons-material/Delete";

export const AnnouncementsPage = () => {
    const initialMailList = [];
    const [mailList, setMailList] = React.useState(initialMailList);
    const [verUser, setVerUser] = React.useState({
        mail:'',
        name:'',
    });


    const handleMailChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        setVerUser({...verUser, mail:event.target.value});
    };

    const handleNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        setVerUser({...verUser, name:event.target.value});
    };

    const handleAddMail = () => {
        if (verUser.name !== '' && verUser.mail !== '') {
            const newUser = {
                mail:verUser.mail,
                name:verUser.name,
            }
            setMailList(mailList.concat(newUser));
        }
        alert('Username and email have to be filled out')
    };

    const handleDeleteUser = (mail) => {
        setMailList(mailList.filter(item => item.mail !== mail));
    }


    return(
        <div>
            <Grid container spacing={2} direction="row" justifyContent="flex-start" alignItems="center">
                <Grid item xs={12}>
                    <h1>Add e-mail as verified: </h1>
                </Grid>
                <Grid item>
                    <TextField
                        label="E-Mail"
                        value={verUser.mail}
                        onChange={handleMailChange}
                    />
                </Grid>
                <Grid item>
                    <TextField
                        label="Name"
                        value={verUser.name}
                        onChange={handleNameChange}
                    />
                </Grid>
                <Grid item>
                    <Button variant="outlined" onClick={handleAddMail}>
                        Add
                    </Button>
                </Grid>
                <Grid item xs={12}>
                    <h1>Currently verified accounts:</h1>
                </Grid>
                <Grid item xs={4}>
                    Email
                </Grid>
                <Grid item xs={2}>
                    Name
                </Grid>
                <Grid item xs={6}></Grid>
                <Grid item xs={12}>
                    <ul>
                        {mailList.map(item => (
                            <Grid container spacing={2} direction="row" justifyContent="flex-start" alignItems="center">
                                <Grid item xs={4}>
                                    {item.mail}
                                </Grid>
                                <Grid item xs={1}>
                                    {item.name}
                                </Grid>
                                <Grid item xs={1}>
                                    <Button onClick={() => {handleDeleteUser(item.mail)}}>
                                        <DeleteIcon/>
                                    </Button>
                                </Grid>
                            </Grid>
                        ))}
                    </ul>
                </Grid>
            </Grid>
        </div>
    );
}