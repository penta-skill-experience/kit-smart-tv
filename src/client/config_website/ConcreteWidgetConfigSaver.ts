import {WidgetConfigSaver} from "../widget/WidgetConfigSaver";

export class ConcreteWidgetConfigSaver implements WidgetConfigSaver {

    save(rawConfig: Object): void {

        // todo write to persistence
        console.log("save widget config:");
        console.log(rawConfig);
    }
}