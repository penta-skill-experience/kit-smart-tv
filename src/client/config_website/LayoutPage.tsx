import * as React from "react";
import MenuItem from '@mui/material/MenuItem';
import FormControl from '@mui/material/FormControl';
import Select from '@mui/material/Select';
import {Button, Grid} from "@mui/material";
import Box from "@mui/material/Box";
import Typography from "@mui/material/Typography";
import LockIcon from '@mui/icons-material/Lock';
import {WidgetListElement} from "./WidgetListElement";
import {WidgetLoader} from "../widget/WidgetLoader";
import Snackbar from "@mui/material/Snackbar";


const widgetLoader = new WidgetLoader();
const widgetList = widgetLoader.getWidgetIds();


export const LayoutPage = ({list, widgetListElement, handleWidgetSelection, handleAddWidget, handleDeleteWidget, handlePosition, handleRawConfigSave, handleLayoutChange, children}) => {

    const [open, setOpen] = React.useState(false);

    const handleClick = () => {
        setOpen(true);
    };

    const handleClose = (event: React.SyntheticEvent | Event, reason?: string) => {
        if (reason === 'clickaway') {
            return;
        }
        setOpen(false);
    }

    return(
        <div>
            <Grid container spacing={2} direction="row" justifyContent="flex-start" alignItems="flex-start">
                <Grid item container spacing={2} direction="row" xs={12}>
                    <Grid item>
                        <Box
                            sx={{
                                width: 150,
                                height: 150,
                                backgroundColor: 'text.primary',
                            }
                            }
                        >
                            <Grid container direction="row" alignItems="center" justifyContent="space-between">
                                <Grid item xs={12}>''</Grid>
                                <Grid item xs={12}>''</Grid>
                                <Grid item  container alignItems="center" justifyContent="center">
                                    <Grid item>
                                        <LockIcon color='error'/>
                                    </Grid>
                                </Grid>
                            </Grid>
                        </Box>

                    </Grid>

                    <Grid item>
                        <Box
                            sx={{
                                width: 150,
                                height: 150,
                                backgroundColor: 'text.primary',
                                }
                            }
                        >
                            <Grid container direction="row" alignItems="center" justifyContent="space-between">
                                <Grid item xs={12}>''</Grid>
                                <Grid item xs={12}>''</Grid>
                                <Grid item  container alignItems="center" justifyContent="center">
                                    <Grid item>
                                        <Typography variant="h6" color='white'>1.top center</Typography>
                                    </Grid>
                                </Grid>
                            </Grid>
                        </Box>

                    </Grid>

                    <Grid item>
                        <Box
                            sx={{
                                width: 150,
                                height: 150,
                                backgroundColor: 'text.primary',
                            }
                            }
                        >
                            <Grid container direction="row" alignItems="center" justifyContent="space-between">
                                <Grid item xs={12}>''</Grid>
                                <Grid item xs={12}>''</Grid>
                                <Grid item  container alignItems="center" justifyContent="center">
                                    <Grid item>
                                        <Typography variant="h6" color='white'>2.top right</Typography>
                                    </Grid>
                                </Grid>
                            </Grid>
                        </Box>

                    </Grid>

                </Grid>

                <Grid item container spacing={2} direction="row">
                    <Grid item>
                        <Box
                            sx={{
                                width: 150,
                                height: 150,
                                backgroundColor: 'text.primary',
                            }
                            }
                        >
                            <Grid container direction="row" alignItems="center" justifyContent="space-between">
                                <Grid item xs={12}>''</Grid>
                                <Grid item xs={12}>''</Grid>
                                <Grid item  container alignItems="center" justifyContent="center">
                                    <Grid item>
                                        <Typography variant="h6" color='white'>3.bottom left</Typography>
                                    </Grid>
                                </Grid>
                            </Grid>
                        </Box>

                    </Grid>

                    <Grid item>
                        <Box
                            sx={{
                                width: 150,
                                height: 150,
                                backgroundColor: 'text.primary',
                            }
                            }
                        >
                            <Grid container direction="row" alignItems="center" justifyContent="space-between">
                                <Grid item xs={12}>''</Grid>
                                <Grid item xs={12}>''</Grid>
                                <Grid item  container alignItems="center" justifyContent="center">
                                    <Grid item>
                                        <Typography variant="h6" color='white'>4.bottom center</Typography>
                                    </Grid>
                                </Grid>
                            </Grid>
                        </Box>

                    </Grid>

                    <Grid item>
                        <Box
                            sx={{
                                width: 150,
                                height: 150,
                                backgroundColor: 'text.primary',
                            }
                            }
                        >
                            <Grid container direction="row" alignItems="center" justifyContent="space-between">
                                <Grid item xs={12}>''</Grid>
                                <Grid item xs={12}>''</Grid>
                                <Grid item  container alignItems="center" justifyContent="center">
                                    <Grid item>
                                        <Typography variant="h6" color='white'>5.bottom right</Typography>
                                    </Grid>
                                </Grid>
                            </Grid>
                        </Box>

                    </Grid>
                </Grid>

                <Grid item xs={12}>
                    <ul>
                        {list.map(item => (
                            <WidgetListElement
                                item={item}
                                handlePosition={handlePosition}
                                handleDeleteWidget={handleDeleteWidget}
                                handleRawConfigSave={handleRawConfigSave}
                            >
                            </WidgetListElement>
                        ))}
                    </ul>
                </Grid>
                <Grid item xs={12}>
                    <FormControl sx={{minWidth: 120}}>
                        <Select
                            value={widgetListElement.widgetNameText}
                            onChange={handleWidgetSelection}
                        >
                            {widgetList.map(item => (
                                <MenuItem value={item}>
                                    {widgetLoader.getWidget(item).getTitle()}
                                </MenuItem>
                            ))}
                        </Select>
                    </FormControl>
                </Grid>
                <Grid item xs={12}>
                    <Button onClick={handleAddWidget} variant="outlined">Add Widget</Button>
                </Grid>
                <Grid item>
                    <Button variant="contained" onClick={() => {
                        handleLayoutChange();
                        handleClick();
                    }}>
                        Save
                    </Button>
                    <Snackbar
                        open={open}
                        autoHideDuration={6000}
                        onClose={handleClose}
                        message={'Changes Saved'}
                    />
                </Grid>
            </Grid>
        </div>
    );
};