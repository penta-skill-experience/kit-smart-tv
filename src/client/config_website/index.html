<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Admin Interface</title>
    <meta content="width=device-width, initial-scale=1" name="viewport">
    <link
            href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap"
            rel="stylesheet"
    />
    <style>
        #root {
            height: 100vh;
            width: 100vw;
        }
    </style>
</head>
<body>
<div id="root">
    <!-- react output will go inside here -->
</div>
</body>
</html>
