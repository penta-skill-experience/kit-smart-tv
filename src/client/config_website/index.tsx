import "./index.css";
import * as ReactDOM from "react-dom";
import * as React from "react";
import {WidgetLoader} from "../widget/WidgetLoader";
import {ConcreteWidgetConfigSaver} from "./ConcreteWidgetConfigSaver";

ReactDOM.render(
    // this is just an example to demonstrate how widgets are instantiated
    // todo: remove this.
    <div>
        <h1>Hello Admin!</h1>
        <div>
            {new WidgetLoader().getWidget("rss-feed").createConfigComponent({url: "www.example.com"}, new ConcreteWidgetConfigSaver())}
        </div>
    </div>,
    document.getElementById("root")
);