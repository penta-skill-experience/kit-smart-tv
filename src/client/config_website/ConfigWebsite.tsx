import Typography from '@mui/material/Typography';
import * as React from "react";
import Tabs from '@mui/material/Tabs';
import Tab from '@mui/material/Tab';
import Box from '@mui/material/Box';
import {PersonalizationPage} from "./PersonalizationPage";
import {LayoutPage} from "./LayoutPage";
import {AdminPage} from "./AdminPage";
import {useState} from "react";
import {SelectChangeEvent} from "@mui/material/Select";
import {LogInPage} from "./LogInPage";
import {Button, Grid} from "@mui/material";
import {AnnouncementsPage} from "./AnnouncementsPage";
import * as emailValidator from "email-validator";


interface TabPanelProps {
    children?: React.ReactNode;
    index: number;
    value: number;
}

function TabPanel(props: TabPanelProps) {
    const {children, value, index} = props;

    return (
        <div
            role="tabpanel"
            hidden={value !== index}
        >
            {value === index && (
                <Box sx={{ p: 3 }}>
                    <Typography>{children}</Typography>
                </Box>
            )}
        </div>
    );
}

export function ConfigWebsite() {
    //state variables and methods for login page
    const password = 'password123';
    const [logInInput, setLogInInput] = useState('');
    const [visible, setVisible] = useState(false);
    const [loggedInStatus, setLoggedInStatus] = useState(false);

    const handleInput = (
        event: React.ChangeEvent<HTMLInputElement>
    ) => {
        setLogInInput(event.target.value);
    };

    const handleLogIn= () => {
        // todo
        // connect with persistence
        if (logInInput === password) {
            setLoggedInStatus(true);
        } else {
            alert('password not correct')
        }
    };

    const handleClickShowPassword = () => {
        setVisible(!visible)
    }

    //state variables and methods for tabs
    const [pageNumber, setPageNumber] = React.useState(0);

    const handleChange = (event: React.SyntheticEvent, newValue: number) => {
        setPageNumber(newValue);
    };

    //state variables and methods for personalization page
    const [colorScheme,setColorScheme] = useState<string | null>(null);
    const [fontSize, setFontSize] = useState<string | null>(null);

    const handleColorSchemeChange = (
        event: React.MouseEvent<HTMLElement>,
        newColorScheme: string | null,
    ) => {
        setColorScheme(newColorScheme);
    };

    const handleFontSizeChange = (
        event: React.MouseEvent<HTMLElement>,
        newFontSize: string | null,
    ) => {
        setFontSize(newFontSize);
    };

    const [selectedLightImage, setSelectedLightImage] = React.useState('');

    const handleLightImageSelect = (event) => {
        setSelectedLightImage(event.target.value);
    };

    const [selectedDarkImage, setSelectedDarkImage] = React.useState('');

    const handleDarkImageSelect = (event) => {
        setSelectedDarkImage(event.target.value);
    };

    //state variables and methods for layout page
    const initialList = [];
    const [counter, setCounter] = useState(1);
    const [list, setList] = React.useState(initialList);
    const [widget, setWidget] = React.useState({
        id:0,
        name:'',
        position:'',
        configurable:false,
        colorSolid:false
    });

    const incrementCounter = () => setCounter(counter + 1);

    const handleWidgetSelection = (event: SelectChangeEvent) => {
        //todo
        //config is not always true
        const updatedValue = {
            id:counter,
            name:event.target.value,
            position:'',
            configurable:true,
            colorSolid:false
        }
        setWidget(updatedValue)
    };

    const handleAddWidget = () => {
        if (widget.name !== '') {
            const newWidget = {
                //todo
                //config is not always true
                id:counter,
                name:widget.name,
                position:'',
                configurable:true,
                colorSolid:false
            }
            setList(list.concat(newWidget));
            incrementCounter();
            console.log('Added widget ' + newWidget.name + ' with id ' + newWidget.id)
        }
    };

    const handleDeleteWidget = (id) => {
        console.log('Widget with id ' + id + ' is removed ')
        setList(list.filter(item => item.id !== id));
    }

    const handlePosition = (id, position) => {
        const newList = list.map((item) => {
            if (item.id === id) {
                const newWidget = { ...item, position: position }
                return newWidget;
            } else {
                return item;
            };
        });

        setList(newList);

    }

    const handleColorSolid = (id, isChecked) => {
        const newList = list.map((item) =>{
            if (item.id === id) {
                const newWidget = {...item, colorSolid: !isChecked}
                return newWidget;
            } else {
                return item;
            };
        });
        setList(newList)
    }

    //state variables and methods for admin password page
    const [newPassword, setNewPassword] = useState('')
    const [oldPassword, setOldPassword] = useState('')

    const handleOldPassword = (
        event: React.ChangeEvent<HTMLInputElement>
    ) => {
        setOldPassword(event.target.value);
    };

    const handleNewPassword = (
        event: React.ChangeEvent<HTMLInputElement>
    ) => {
        setNewPassword(event.target.value);
    };

    //todo
    //Connect with persistence
    const handlePasswordChange = () => {
        alert('Old Password is ' + oldPassword + ' New Password is ' + newPassword)
    };

    const handleChangeSave = () => {
        if (colorScheme === null || fontSize === null) {
            alert('Color scheme and font size must be chosen')
        }
    };

    //state variables and methods for announcements page
    const initialMailList = [];
    const [mailList, setMailList] = React.useState(initialMailList);
    const [verUser, setVerUser] = React.useState({
        mail:'',
        name:'',
    });

    const handleMailChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        setVerUser({...verUser, mail:event.target.value});
    };

    const handleNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        setVerUser({...verUser, name:event.target.value});
    };

    const handleAddMail = () => {
        const alreadyExists = mailList.some(item => verUser.mail === item.mail);
        if (!emailValidator.validate(verUser.mail)) {
            alert('E-Mail does not exist')
            return;
        }
        if (alreadyExists === true) {
            alert('This E-Mail already exists')
            return;;
        }
        if (verUser.name !== '' && verUser.mail !== '') {
            const newUser = {
                mail:verUser.mail,
                name:verUser.name,
            }
            setMailList(mailList.concat(newUser));
            return;
        }
        alert('Username and email have to be filled out')
    };

    const handleDeleteUser = (mail) => {
        setMailList(mailList.filter(item => item.mail !== mail));
    }

    //state variable for log out

    const handleLogout = () => {
        setLoggedInStatus(false);
    }

    function renderConfigWebsite() {
        if(loggedInStatus === false) {
            return (
                <LogInPage
                    logInInput={logInInput}
                    visible={visible}
                    handleInput={handleInput}
                    handleLogIn={handleLogIn}
                    handleClickShowPassword={handleClickShowPassword}
                >
                </LogInPage>
            );
        } else {
            return (
                <div>
                    <Box sx={{
                        border: 1,
                        backgroundColor:'text.primary',}
                    }>
                        <Grid container spacing={2} direction="row" alignItems="center">
                            <Grid item xs={2}></Grid>
                            <Grid item container xs={8} alignItems="center" justifyContent="center">
                                <Grid item container alignItems="center" justifyContent="center">
                                    <Grid item>
                                        <Typography variant="h6" color='white'>Admin Interface</Typography>
                                    </Grid>
                                </Grid>
                            </Grid>
                            <Grid item container xs={2} direction="column">
                                <Button onClick={handleLogout}>
                                    Log Out
                                </Button>
                            </Grid>
                        </Grid>
                    </Box>

                    <Box sx={{ borderBottom: 1, borderColor: 'divider' }}>
                        <Tabs value={pageNumber} onChange={handleChange}>
                            <Tab label="Personalization"/>
                            <Tab label="Layout"/>
                            <Tab label="Password"/>
                            <Tab label="Announcements"/>
                        </Tabs>
                    </Box>
                    <TabPanel value={pageNumber} index={0}>
                        <PersonalizationPage
                            colorScheme={colorScheme}
                            fontSize={fontSize}
                            handleColorSchemeChange={handleColorSchemeChange}
                            handleFontSizeChange={handleFontSizeChange}
                            selectedLightImage={selectedLightImage}
                            selectedDarkImage={selectedDarkImage}
                            handleLightImageSelect={handleLightImageSelect}
                            handleDarkImageSelect={handleDarkImageSelect}
                        >
                        </PersonalizationPage>

                    </TabPanel>
                    <TabPanel value={pageNumber} index={1}>
                        <LayoutPage
                            list={list}
                            widget={widget}
                            handleWidgetSelection={handleWidgetSelection}
                            handleAddWidget={handleAddWidget}
                            handleDeleteWidget={handleDeleteWidget}
                            handlePosition={handlePosition}
                            handleColorSolid={handleColorSolid}
                        >
                        </LayoutPage>
                    </TabPanel>
                    <TabPanel value={pageNumber} index={2}>
                        <AdminPage
                            oldPassword={oldPassword}
                            newPassword={newPassword}
                            handleOldPassword={handleOldPassword}
                            handleNewPassword={handleNewPassword}
                            handlePasswordChange={handlePasswordChange}
                            handleChangeSave={handleChangeSave}
                        >
                        </AdminPage>
                    </TabPanel>
                    <TabPanel value={pageNumber} index={3}>
                        <AnnouncementsPage
                            mailList={mailList}
                            verUser={verUser}
                            handleMailChange={handleMailChange}
                            handleNameChange={handleNameChange}
                            handleAddMail={handleAddMail}
                            handleDeleteUser={handleDeleteUser}
                        >
                        </AnnouncementsPage>
                    </TabPanel>
                </div>
            );
        }

    }

    return(<div>
            {renderConfigWebsite()}
        </div>
    );
}