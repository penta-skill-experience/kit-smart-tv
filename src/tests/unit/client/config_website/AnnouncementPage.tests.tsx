import {shallow, configure} from "enzyme";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import * as React from "react";
import toJson from "enzyme-to-json";
import {AnnouncementsPage} from "../../../../client/config_website/AnnouncementsPage";


configure({adapter: new Adapter()});

describe("Announcement page tests", () => {
    let wrapper;
    const list = [];
    const verUserListElement = {
        mail:'',
        name:'',
        verUser:null,
    };
    const method = () => {
    };



    beforeEach(() => {
        wrapper = shallow(
            <AnnouncementsPage
                mailList={list}
                verUser={verUserListElement}
                handleMailChange={method}
                handleNameChange={method}
                handleAddMail={method}
                handleDeleteUser={method}
                handleVerUserList={method}
            >
            </AnnouncementsPage>
        );
    });

    test("snapshot test for announcement page", async () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

});