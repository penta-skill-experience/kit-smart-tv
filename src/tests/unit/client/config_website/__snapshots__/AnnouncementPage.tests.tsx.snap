// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Announcement page tests snapshot test for announcement page 1`] = `
<div>
  <ForwardRef(Grid)
    alignItems="center"
    container={true}
    direction="row"
    justifyContent="flex-start"
    spacing={2}
  >
    <ForwardRef(Grid)
      item={true}
      xs={12}
    >
      <h1>
        Add e-mail as verified: 
      </h1>
    </ForwardRef(Grid)>
    <ForwardRef(Grid)
      item={true}
    >
      <ForwardRef(TextField)
        label="E-Mail"
        onChange={[Function]}
        value=""
      />
    </ForwardRef(Grid)>
    <ForwardRef(Grid)
      item={true}
    >
      <ForwardRef(TextField)
        label="Name"
        onChange={[Function]}
        value=""
      />
    </ForwardRef(Grid)>
    <ForwardRef(Grid)
      item={true}
    >
      <ForwardRef(Button)
        onClick={[Function]}
        variant="outlined"
      >
        Add
      </ForwardRef(Button)>
      <ForwardRef(Snackbar)
        autoHideDuration={6000}
        message="E-Mail does not exist"
        onClose={[Function]}
        open={false}
      />
      <ForwardRef(Snackbar)
        autoHideDuration={6000}
        message="This E-Mail already exists"
        onClose={[Function]}
        open={false}
      />
      <ForwardRef(Snackbar)
        autoHideDuration={6000}
        message="Username and email have to be filled out"
        onClose={[Function]}
        open={false}
      />
    </ForwardRef(Grid)>
    <ForwardRef(Grid)
      item={true}
      xs={12}
    >
      <h1>
        Currently verified accounts:
      </h1>
    </ForwardRef(Grid)>
    <ForwardRef(Grid)
      item={true}
      xs={4}
    >
      Email
    </ForwardRef(Grid)>
    <ForwardRef(Grid)
      item={true}
      xs={2}
    >
      Name
    </ForwardRef(Grid)>
    <ForwardRef(Grid)
      item={true}
      xs={6}
    />
    <ForwardRef(Grid)
      item={true}
      xs={12}
    >
      <ul>
        <ForwardRef(Grid)
          alignItems="center"
          container={true}
          direction="row"
          justifyContent="flex-start"
          spacing={2}
        >
          <ForwardRef(Grid)
            item={true}
            xs={4}
          >
            testEmail@gmail.com
          </ForwardRef(Grid)>
          <ForwardRef(Grid)
            item={true}
            xs={1}
          >
            Bob
          </ForwardRef(Grid)>
          <ForwardRef(Grid)
            item={true}
            xs={1}
          >
            <ForwardRef(Button)
              onClick={[Function]}
            >
              <Memo(ForwardRef(DeleteIcon)) />
            </ForwardRef(Button)>
          </ForwardRef(Grid)>
        </ForwardRef(Grid)>
      </ul>
    </ForwardRef(Grid)>
    <ForwardRef(Grid)
      item={true}
      xs={12}
    >
      <ForwardRef(Button)
        onClick={[Function]}
        variant="outlined"
      >
        Save Changes
      </ForwardRef(Button)>
      <ForwardRef(Snackbar)
        autoHideDuration={6000}
        message="Verified Users Saved"
        onClose={[Function]}
        open={false}
      />
      <ForwardRef(Snackbar)
        autoHideDuration={6000}
        message="Session expired"
        onClose={[Function]}
        open={false}
      />
    </ForwardRef(Grid)>
  </ForwardRef(Grid)>
</div>
`;
